test/unittest/tests/test_string_match.cpp

```diff
@@ -0,0 +1,121 @@
+/*-
+ * Copyright (c) 2014-present MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *	All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+#include <catch2/catch.hpp>
+#include "wt_internal.h"
+
+TEST_CASE("Test string matching macro: WT_STRING_MATCH", "[STRING_MATCH]")
+{
+    WT_CONFIG_ITEM green, empty;
+    int i;
+    const char *empty_str, *g_str, *greed_str, *green_str, *greener_str;
+    const char *null_str, *something_str;
+
+    /* The "right side" of the match */
+    green.str = "green";
+    green.len = 5;
+    empty.str = "";
+    empty.len = 0;
+
+    /* The "left side" of the match */
+    empty_str = "";
+    g_str = "g";
+    greed_str = "greed";
+    green_str = "green";
+    greener_str = "greener";
+    null_str = NULL;
+    something_str = "greener";
```

Comment:
A comment here for why `something` needs to be "greener" instead of "something" since it isn't following the pattern.

Reply:
Thanks, I changed the string to `"something"`.